File: hw/cañarte/u3/HW29Composite&TemplateMethod/Python/Template/controller/Tea.py
import sys
import os
# Saray Cañarte, codecreators, DCCO - ESPE

sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
from controller.CoffeeinBeverage import CoffeeinBeverage

class Tea(CoffeeinBeverage):
    def brew(self):
        print("Steeping the tea")

    def add_condiments(self):
        print("Adding honey")

    def wants_condiments(self):
        answer = self.get_user_input("Would you like honey with your tea (y/n)? ")
        return answer.lower().startswith('y')

    def get_user_input(self, prompt):
        try:
            return input(prompt)
        except Exception as e:
            print("An error occurred while trying to read your input. Please try again.")
            return "no"

File: hw/CEDENOR/U3/HW29Composite&TemplateMethod/Java/BeverageApp/src/main/java/ec/edu/espe/beverageapp/controller/Tea.java
package ec.edu.espe.beverageapp.controller;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

/**
 *
 * @author Nahomi Cedeño, codecreators, DCCO - ESPE
 */
public class Tea extends CoffeeinBeverage {

    @Override
    void brew() {
        System.out.println("Steeping the tea");
    }

    @Override
    void addCondiments() {
        System.out.println("Adding honey");
    }

    @Override
    boolean wantsCondiments() {
        String answer = getUserInput();
        if (answer.toLowerCase().startsWith("y")){
            return true;
        }else{
            return false;
        }
    }
    
    private String getUserInput() {
        String answer = null;
        System.out.println("Would you like honey with your tea (y/n)?");

        try {
            BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
            answer = in.readLine();  
        } catch (IOException e) {
            System.out.println("An error occurred while trying to read your input. Please try again.");
            e.printStackTrace(); 
        }

        return answer;
    }

}
